pkg/internal/controller/controller.go

```diff
@@ -86,8 +88,8 @@ type watchDescription struct {
 }
 
 // Reconcile implements reconcile.Reconciler
-func (c *Controller) Reconcile(r reconcile.Request) (reconcile.Result, error) {
-	return c.Do.Reconcile(r)
+func (c *Controller) Reconcile(ctx context.Context, r reconcile.Request) (reconcile.Result, error) {
```

Comment:
It's not overly clear to me when this `Reconcile()` method may be called as opposed to the logic in `reconcileHandler()`, should it also do similar enrichment of the log values and inject the log into the context here as well?